### ISSUE TITLE
N64 Pairing Issues

### ISSUE THREAD

**Issue by s3anami** [darthcloud/BlueRetro#276]
I have an N64 with a UltraHDMI
I tried pairing a Xbox One controller with it and it seems the only manner I can get it working is to have the Xbox in pair mode before the console powers on and it will sync immediately.
However if I try to sync any controllers after the console has been on for a bit it is unsuccessful.
I am using this ESP32 revision: https://www.amazon.com/dp/B0811LGWY2?psc=1&ref=ppx_yo2ov_dt_b_product_details
The N64 cable is soldered directly to the 3.3V, 19 and GND pins on the board.

**Comment by darthcloud** [comment]
Maybe you got a rogue BT HID device at home that connect automatically to BlueRetro.

Try to get the logs:
https://github.com/darthcloud/BlueRetro/wiki/Getting-BlueRetro-debug-logs-via-Serial-port-Windows-10

Will log running, reset ESP32 and wait a minute, do not connect any controller.

**Comment by s3anami** [comment]
Logs after reset for couple mins i see 1 error  not sure  what it is
▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒▒I (244) cpu_start: Pro cpu up.
I (244) cpu_start: Single core mode
I (252) cpu_start: Pro cpu start user code
I (252) cpu_start: cpu freq: 240000000 Hz
I (252) cpu_start: Application information:
I (252) cpu_start: Project name:     BlueRetro
I (253) cpu_start: App version:      v1.5.1 n64_spiffs
I (254) cpu_start: Compile time:     Mar 30 2022 00:06:31
I (254) cpu_start: ELF file SHA256:  a415d94d85b306e9...
I (255) cpu_start: ESP-IDF:          v5.0-dev-1426-ge899edd793
I (256) cpu_start: App cpu up.
I (257) heap_init: Initializing. RAM available for dynamic allocation:
I (257) heap_init: At 3FFAFF10 len 000000F0 (0 KiB): DRAM
I (258) heap_init: At 3FFB6388 len 00001C78 (7 KiB): DRAM
I (259) heap_init: At 3FFB9A20 len 00004108 (16 KiB): DRAM
I (259) heap_init: At 3FFD02A8 len 0000FD58 (63 KiB): DRAM
I (260) heap_init: At 3FFE0440 len 00003AE0 (14 KiB): D/IRAM
I (261) heap_init: At 3FFE4350 len 00003A48 (14 KiB): D/IRAM
I (262) heap_init: At 3FFE7E28 len 000181D8 (96 KiB): D/IRAM
I (263) heap_init: At 40078000 len 00008000 (32 KiB): IRAM
I (263) heap_init: At 4009E588 len 00001A78 (6 KiB): IRAM
I (264) heap_init: At 3FF80000 len 00002000 (8 KiB): RTCRAM
I (266) spi_flash: detected chip: generic
I (267) spi_flash: flash io: dio
I (267) esp_core_dump_uart: Init core dump to UART
I (267) esp_core_dump_uart: Init core dump to UART
I (268) cpu_start: Starting scheduler on PRO CPU.
# bt_host_load_bdaddr_from_file: No BDADDR on SD. Using ESP32's MAC
I (357) BTDM_INIT: BT controller compile version [33445ec]
I (357) system_api: Base MAC address is not set
I (357) system_api: read default base MAC address from EFUSE
I (367) phy_init: phy_version 4670,719f9f6,Feb 18 2021,17:07:07
# bt_hci_cmd_reset
I (777) gpio: GPIO[0]| InputEn: 1| OutputEn: 0| OpenDrain: 0| Pullup: 1| Pulldown: 0| Intr:0
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_read_local_features
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_read_local_version_info
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_read_bd_addr
# BT_HCI_EVT_CMD_COMPLETE
# local_bdaddr: 30:C6:F7:29:00:92
# bt_hci_cmd_read_buffer_size
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_read_class_of_device
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_read_local_name
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_read_voice_setting
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_read_num_supported_iac
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_read_current_iac_lap
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_set_event_filter
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_write_conn_accept_timeout
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_read_supported_commands
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_write_ssp_mode
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_write_inquiry_mode
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_read_inquiry_rsp_tx_pwr_lvl
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_read_local_ext_features
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_read_stored_link_key
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_read_page_scan_activity
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_read_page_scan_type
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_write_le_host_supp
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_le_read_wl_size
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_le_clear_wl
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_le_add_dev_to_wl
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_le_add_dev_to_wl
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_delete_stored_link_key
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_write_class_of_device
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_write_local_name
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_set_event_filter
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_set_event_filter
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_write_auth_enable
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_set_event_mask
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_write_page_scan_activity
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_write_inquiry_scan_activity
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_write_page_scan_type
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_write_page_scan_timeout
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_write_hold_mode_act
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_write_scan_enable
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_write_default_link_policy
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_le_read_buffer_size
# BT_HCI_EVT_CMD_COMPLETE
# bt_att_set_le_max_mtu 247
# bt_hci_cmd_le_set_adv_param
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_le_set_adv_data
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_le_set_scan_rsp_data
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_le_set_adv_enable
# BT_HCI_EVT_CMD_COMPLETE
# bt_hci_cmd_le_set_scan_enable
# bt_hci_cmd_le_set_scan_param_active
# bt_hci_cmd_le_set_scan_enable
# bt_hci_cmd_periodic_inquiry
# BT_HCI_EVT_CMD_COMPLETE
**# opcode: 0x200C error: 0x0C retry: 0**
# bt_hci_cmd_le_set_scan_enable
# bt_hci_cmd_le_set_scan_param_active
# bt_hci_cmd_le_set_scan_enable
# bt_hci_cmd_periodic_inquiry
# BT_HCI_EVT_CMD_COMPLETE
# BT_HCI_EVT_CMD_COMPLETE
# BT_HCI_EVT_CMD_COMPLETE
# BT_HCI_EVT_CMD_COMPLETE
# BT_HCI_EVT_CMD_COMPLETE
# BT_HCI_EVT_CMD_COMPLETE
# BT_HCI_EVT_CMD_COMPLETE
**# opcode: 0x0403 error: 0x0C retry: 0
# mc_restore: restore failed! cnt: 22 File size:90112
App main at 40083058
App init at 40083068
APP_CPU RESET: 0
APP_CPU CLKGATE: 0
APP_CPU STALL: 0
APP_CACHE_CTRL: 00000030
APP CPU STACK PTR: 3FFDFAF4
# Hardcoded system : 15: N64**
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_INQUIRY_COMPLETE
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_INQUIRY_COMPLETE
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_INQUIRY_COMPLETE
# BT_HCI_EVT_INQUIRY_COMPLETE
# BT_HCI_EVT_INQUIRY_COMPLETE
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_INQUIRY_COMPLETE
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_INQUIRY_COMPLETE
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_INQUIRY_COMPLETE
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_INQUIRY_COMPLETE
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_INQUIRY_COMPLETE
# BT_HCI_EVT_INQUIRY_COMPLETE
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_INQUIRY_COMPLETE
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_INQUIRY_COMPLETE
# BT_HCI_EVT_INQUIRY_COMPLETE
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_INQUIRY_COMPLETE
# BT_HCI_EVT_INQUIRY_COMPLETE
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_INQUIRY_COMPLETE
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_INQUIRY_COMPLETE
# BT_HCI_EVT_INQUIRY_COMPLETE
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_INQUIRY_COMPLETE
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_INQUIRY_COMPLETE
# BT_HCI_EVT_INQUIRY_COMPLETE
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_INQUIRY_COMPLETE
# BT_HCI_EVT_INQUIRY_COMPLETE
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_INQUIRY_COMPLETE
# BT_HCI_EVT_INQUIRY_COMPLETE
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_INQUIRY_COMPLETE
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_INQUIRY_COMPLETE
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_INQUIRY_COMPLETE
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_LE_ADVERTISING_REPORT
# BT_HCI_EVT_INQUIRY_COMPLETE
# BT_HCI_EVT_INQUIRY_COMPLETE

**Comment by s3anami** [comment]
BT_HCI_EVT_INQUIRY_COMPLETE
 BT_HCI_EVT_LE_ADVERTISING_REPORT
 BT_HCI_EVT_LE_ADVERTISING_REPORT
 BT_HCI_EVT_LE_ADVERTISING_REPORT
 BT_HCI_EVT_LE_ADVERTISING_REPORT
 BT_HCI_EVT_LE_ADVERTISING_REPORT
 BT_HCI_EVT_LE_ADVERTISING_REPORT
 BT_HCI_EVT_LE_ADVERTISING_REPORT
 BT_HCI_EVT_LE_ADVERTISING_REPORT
 BT_HCI_EVT_INQUIRY_COMPLETE
 BT_HCI_EVT_LE_ADVERTISING_REPORT
 BT_HCI_EVT_LE_ADVERTISING_REPORT
 BT_HCI_EVT_INQUIRY_COMPLETE
 BT_HCI_EVT_INQUIRY_COMPLETE
 BT_HCI_EVT_INQUIRY_COMPLETE
 BT_HCI_EVT_LE_ADVERTISING_REPORT
 bt_hci_cmd_le_set_scan_enable
 bt_hci_cmd_le_set_adv_disable
 bt_hci_cmd_le_create_conn
 LE ADV dev: 0 type: 0 bdaddr: 01 - E1:CE:4A:88:DA:20
 BT_HCI_EVT_LE_ADVERTISING_REPORT
 BT_HCI_EVT_CMD_COMPLETE
 BT_HCI_EVT_CMD_COMPLETE
 BT_HCI_EVT_CMD_STATUS
 BT_HCI_EVT_LE_CONN_COMPLETE
 dev: 0 acl_handle: 0x0000
 bt_smp_pairing_req
 bt_hci_cmd_le_set_scan_enable
 bt_hci_cmd_le_set_scan_param_active
 bt_hci_cmd_le_set_scan_enable
 BT_HCI_EVT_CMD_COMPLETE
 opcode: 0x200C error: 0x0C retry: 0
 BT_HCI_EVT_CMD_COMPLETE
 BT_HCI_EVT_CMD_COMPLETE
 BT_HCI_EVT_DISCONN_COMPLETE
 DISCONN from dev: 0
 bt_hci_cmd_le_set_scan_enable
 bt_hci_cmd_le_set_scan_param_active
 bt_hci_cmd_le_set_scan_enable
 bt_hci_cmd_periodic_inquiry
 bt_hci_cmd_le_set_adv_enable
 BT_HCI_EVT_CMD_COMPLETE
 BT_HCI_EVT_CMD_COMPLETE
 BT_HCI_EVT_CMD_COMPLETE
 BT_HCI_EVT_CMD_COMPLETE
 opcode: 0x0403 error: 0x0C retry: 0
 BT_HCI_EVT_CMD_COMPLETE
 BT_HCI_EVT_LE_ADVERTISING_REPORT
 BT_HCI_EVT_LE_ADVERTISING_REPORT
 bt_hci_cmd_le_set_scan_enable
 bt_hci_cmd_le_set_adv_disable
 bt_hci_cmd_le_create_conn
 LE ADV dev: 0 type: 0 bdaddr: 01 - E1:CE:4A:88:DA:20
 BT_HCI_EVT_LE_ADVERTISING_REPORT
 BT_HCI_EVT_CMD_COMPLETE
 BT_HCI_EVT_CMD_COMPLETE
 BT_HCI_EVT_CMD_STATUS
 BT_HCI_EVT_LE_CONN_COMPLETE
 dev: 0 acl_handle: 0x0000
 bt_smp_pairing_req
 bt_hci_cmd_le_set_scan_enable
 bt_hci_cmd_le_set_scan_param_active
 bt_hci_cmd_le_set_scan_enable
 BT_HCI_EVT_CMD_COMPLETE
 opcode: 0x200C error: 0x0C retry: 0
 BT_HCI_EVT_CMD_COMPLETE
 BT_HCI_EVT_CMD_COMPLETE
 BT_HCI_EVT_LE_ADVERTISING_REPORT
 BT_HCI_EVT_LE_ADVERTISING_REPORT
 BT_HCI_EVT_DISCONN_COMPLETE
 DISCONN from dev: 0
 bt_hci_cmd_le_set_scan_enable
 bt_hci_cmd_le_set_scan_param_active
 bt_hci_cmd_le_set_scan_enable
 bt_hci_cmd_periodic_inquiry
 bt_hci_cmd_le_set_adv_enable
 BT_HCI_EVT_CMD_COMPLETE
 BT_HCI_EVT_CMD_COMPLETE
 BT_HCI_EVT_CMD_COMPLETE
 BT_HCI_EVT_CMD_COMPLETE
 opcode: 0x0403 error: 0x0C retry: 0
 BT_HCI_EVT_CMD_COMPLETE
 BT_HCI_EVT_LE_ADVERTISING_REPORT
 BT_HCI_EVT_LE_ADVERTISING_REPORT
 BT_HCI_EVT_LE_ADVERTISING_REPORT
 BT_HCI_EVT_INQUIRY_COMPLETE
 BT_HCI_EVT_LE_ADVERTISING_REPORT
 BT_HCI_EVT_LE_ADVERTISING_REPORT
 BT_HCI_EVT_LE_ADVERTISING_REPORT
 BT_HCI_EVT_LE_ADVERTISING_REPORT
 BT_HCI_EVT_LE_ADVERTISING_REPORT
 BT_HCI_EVT_LE_ADVERTISING_REPORT
 BT_HCI_EVT_LE_ADVERTISING_REPORT
 BT_HCI_EVT_LE_ADVERTISING_REPORT
 BT_HCI_EVT_LE_ADVERTISING_REPORT
 BT_HCI_EVT_LE_ADVERTISING_REPORT
 BT_HCI_EVT_LE_ADVERTISING_REPORT
 BT_HCI_EVT_INQUIRY_COMPLETE
 BT_HCI_EVT_LE_ADVERTISING_REPORT
 BT_HCI_EVT_LE_ADVERTISING_REPORT
 BT_HCI_EVT_INQUIRY_COMPLETE

**Comment by darthcloud** [comment]
Put the log in a txt file and upload the file.

Copy pasting directly make it hard to follow.

**Comment by darthcloud** [comment]
Are connecting a controller while getting the logs?

If not you got a device trying to connect with following MAC:
E1:CE:4A:88:DA:20

**Comment by s3anami** [comment]
Yes that was after waiting 20 minutes and the only time it came up. Could it still be causing issues even if that was the only time i have seen in the logs

**Comment by s3anami** [comment]
https://pastebin.com/9pbc8uUL
you can dl for txt file here

**Comment by s3anami** [comment]
> Are connecting a controller while getting the logs?
> 
> If not you got a device trying to connect with following MAC: E1:CE:4A:88:DA:20

does the opcode: 0x0403 error: 0x0C retry: 0
mc_restore: restore failed! cnt: 22 File size:90112 

mean anything

**Comment by darthcloud** [comment]
error 0x0C is normal.

If the mc restore (memory card) fail all the time maybe try with a genuine devkitc from espressif from mouser or digikey.

**Comment by s3anami** [comment]
> error 0x0C is normal.
> 
> If the mc restore (memory card) fail all the time maybe try with a genuine devkitc from espressif from mouser or digikey.

It does happen everytime, I haven't set up a memory card at all is it a config issue on my end or something? Or reflash or something

**Comment by s3anami** [comment]
i flashed another esp32 from the batch and don't get the mc error, so that one is probably bad i guess

**Comment by darthcloud** [comment]
So no pairing issue with the new ESP32?

**Comment by s3anami** [comment]
yes, the other one was bad, guess clone quality is awful
